easyrsa3/easyrsa

```diff
@@ -573,6 +577,114 @@ Your newly created PKI dir is: $EASYRSA_PKI
 	return 0
 } # => init_pki()
 
+# Copy data-files from various sources
+install_data_to_pki () {
+#
+# This function is here to explicitly copy data-files to the PKI.
+# During 'init-pki' this is the new default.
+# During all other functions these requirements are tested for and
+# files will be copied to the PKI, if they do not already exist there.
+#
+# One of the reasons for this change is to make packing EasyRSA work.
+# This function searches favoured and then common 'areas' for the
+#   EasyRSA data-files(A):
+#     'openssl-easyrsa.cnf' 'x509-types':(folder).
+#
+# These files MUST be found in at least one location and will be copied
+# to the current PKI, if they do not already exist there.
+#
+#
+# Other EasyRSA data-files(B): it is not crucial that these are found
+# but if they are then they are also copied to the PKI.
+#   'vars' 'vars.example'
+#
+#
+# For 'vars' consideration must be given to:
+#   "Where the user expects to find vars!"
+#
+# Currently, *if* 'vars' is copied to the PKI then the PKI 'vars' will take
+# priority over './vars'. But it will not be updated if './vars' is changed.
+#
+# Copying 'vars' to the PKI is complicated, code is included but DISABLED.
```

Comment:
I do agree ..

However, moving all PKI related files into said PKI is where EasyRSA is trying to get.  It is just `vars` is going to be tricky.

Reply:
Isn't `vars` optional?
If it's not found the default values are loaded from within `easyrsa`, or am I wrong?

Reply:
I would be in favour of removing `./vars` altogether and moving `vars` into the PKI at `init-pki`, so there is never any external-to-PKI `vars`, at all.

Reply:
> Isn't `vars` optional? If it's not found the default values are loaded from within `easyrsa`, or am I wrong?

You are absolutely correct.  EasyRSA **wants** to change this default.

Reply:
The problem is, old PKIs will have either nothing or **fixed** `./vars`.

Reply:
The **real** problem is: Once `pki/vars` exists it will _take priority_ and that is a **breaking** change.

Reply:
This patch-set gets things in place so that a more invasive change is less objectionable, in time.

Reply:
There may be a way to infiltrate all new PKIs so that `./vars` can be copied to the PKI but only if it does not `grep` for a specific string.

@Prouflon you are to blame `;-)`

Reply:
I absolutely support the notion of a warning, that appears if a `./vars` file ist detected, that tells the user that this file is _not_ read and that this default behaviour has _changed_.
We can expect users to move old `vars` files into the `/pki` on their own when the time comes, right?